### PR TITLE
fix for PT story Create CFH endpoint

### PR DESCRIPTION
**Pull request by casperjureskog**
**Fix for PT Story:**  https://www.pivotaltracker.com/story/show/143863917

**Changes proposed in this PR:**
*adds  change inheritance to  api  for the controller
*adds  change name of column to urgent and start value to false

### CODE REVIEW

File: db/migrate/20170419073022_create_cfhs.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ def change
     create_table :cfhs do |t|
       t.string :title
       t.string :message
-      t.boolean :status
+      t.boolean :urgent, default: false
```
Comment (tochman) [inline db/migrate/20170419073022_create_cfhs.rb]: I don't think you can change an old migration. You should create a new one
```
rails g migration modify_help_requests
```
```ruby
remove_column :help_requests, :status
add_column :help_requests, :urgent, :boolean, default: false
```

**Review by tochman — CHANGES_REQUESTED** [review]

File: spec/models/help_request_spec.rb [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
     it { is_expected.to have_db_column :message }
-    it { is_expected.to have_db_column :status }
+    it { is_expected.to have_db_column :urgent }
   end
 
   describe 'Validations' do
     it { is_expected.to validate_presence_of :title }
     it { is_expected.to validate_presence_of :message }
-    it { is_expected.to validate_presence_of :status }
+    it { is_expected.to validate_presence_of :urgent }
```
Comment (tochman) [inline spec/models/help_request_spec.rb]: Still not sure about this validation

**Comment by tochman** [comment]
I would like to see a sad path spec for the create method now that you have an error block.

File: spec/models/help_request_spec.rb (hunk 2 above)
Comment (casperjureskog) [reply spec/models/help_request_spec.rb]: ok should we remove it ?

Comment (tochman) [reply spec/models/help_request_spec.rb]: Test it using your client or by creating a request spec where :urgent is not being passed in (the way the client behaves at the moment) It's your call.

File: spec/requests/api/v1/helprequest_spec.rb [hunk 3]
```diff
@@ -7,13 +7,18 @@
...
-      response.code.should eq "204"
-      expect(HelpRequest.where(title: "Burning", message: 'Pants is burning',status: true )).to be_present
+      expect(response.code).to eq '200'
+      object = HelpRequest.find_by(title: 'Burning', message: 'Pants is burning', urgent: true )
+      expect(object).to be_persisted
     end
+    it 'should render code 400 on failure' do
+     post '/api/v1/help_requests'
+     expect(response.code).to eq '400'
+   end
```
Comment (tochman) [inline spec/requests/api/v1/helprequest_spec.rb]: Well, I would test for the content of the response since you are returning that.

Comment (tochman) [reply spec/requests/api/v1/helprequest_spec.rb]: Also, mind code formatting.

**Review by tochman — APPROVED** [review]

**Pull request merged**